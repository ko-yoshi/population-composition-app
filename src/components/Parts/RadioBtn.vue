<template>
  <label class="radio-btn">
    <input type="radio" :name="props.name" :checked="props.checked" />
    {{ props.label }}
  </label>
</template>
<script lang="ts">
  type Props = { label: string; name?: string; checked?: boolean }
</script>
<script setup lang="ts">
  const props = withDefaults(defineProps<Props>(), {
    label: '',
    name: '',
    checked: false,
  })
</script>
<style lang="scss" scoped>
  .radio-btn {
    position: relative;
    display: flex;
    gap: 0 0.5em;
    align-items: center;
    max-width: 150px;
    padding: 0.5em 0.7em;
    margin-bottom: 0.4em;
    font-size: 12px;
    cursor: pointer;
    background-color: #d0255026;
    border: 1px solid #d02550;
    border-radius: 3px;
  }

  .radio-btn:has(:checked) {
    color: #fff;
    background-color: #d02550;
  }

  .radio-btn::before,
  .radio-btn:has(:checked)::after {
    content: none;
    border-radius: 50%;
  }

  .radio-btn::before {
    width: 14px;
    height: 14px;
    background-color: #fff;
  }

  .radio-btn:has(:checked)::after {
    position: absolute;
    top: 50%;
    left: calc(7px + 0.7em);
    width: 7px;
    height: 7px;
    background-color: #d02550;
    transform: translate(-50%, -50%);
  }

  .radio-btn input {
    display: none;
  }
</style>
