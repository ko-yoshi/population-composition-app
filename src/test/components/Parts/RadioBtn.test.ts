import { mount } from '@vue/test-utils'
import { expect, test, describe } from 'vitest'
import RadioBtn from '~/components/Parts/RadioBtn.vue'

describe('PartsCheckboxBtn', () => {
  const wrapper = mount(RadioBtn, {
    props: {
      label: 'テスト',
    },
  })
  const label = wrapper.find('label')

  test('props確認 labelのみ設定', () => {
    expect(wrapper.props()).toEqual({ label: 'テスト', name: '', checked: false })
    expect(label.text()).toEqual('テスト')
  })

  test('props確認 全て設定', async () => {
    await wrapper.setProps({
      label: 'テスト2',
      name: 'name-radio',
      checked: true,
    })
    expect(wrapper.props()).toEqual({ label: 'テスト2', checked: true, name: 'name-radio' })
    expect(label.text()).toEqual('テスト2')
  })

  test('snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
