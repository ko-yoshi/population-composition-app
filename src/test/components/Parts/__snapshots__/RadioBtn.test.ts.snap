// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PartsCheckboxBtn > snapshot 1`] = `"<label data-v-63b2aec9=\\"\\" class=\\"radio-btn\\"><input data-v-63b2aec9=\\"\\" type=\\"radio\\" name=\\"name-radio\\"> テスト2</label>"`;
